File: frontend/src/api/aggregate.ts
// Copyright (c) 2024-2026 Morris A. Aguilar. All Rights Reserved.
// Licensed under the Apache License, Version 2.0.
import { apiFetch } from "./client";
import type { AggregateResponse } from "../types/aggregate";

export function getAggregate(model?: string): Promise<AggregateResponse> {
  const params = model ? `?model=${encodeURIComponent(model)}` : "";
  return apiFetch<AggregateResponse>(`/aggregate${params}`);
}

File: server/routers/aggregate.py
# Copyright (c) 2024-2026 Morris A. Aguilar. All Rights Reserved.
# Licensed under the Apache License, Version 2.0.
from fastapi import APIRouter, Query

from server.config import settings
from server.models import AggregateResponse
from server.services import aggregate_service

router = APIRouter()


@router.get("/aggregate", response_model=AggregateResponse)
def get_aggregate(model: str = Query(default=None)):
    model_name = model or settings.default_model
    return aggregate_service.compute_aggregate(model_name, settings.output_path)
